node/network/protocol/src/peer_set.rs

```diff
@@ -57,9 +57,12 @@ impl PeerSet {
 				notifications_protocol: protocol,
 				max_notification_size,
 				set_config: sc_network::config::SetConfig {
-					// we want our gossip subset to always include reserved peers
-					in_peers: super::MIN_GOSSIP_PEERS as u32 / 2,
-					out_peers: 0,
+					// we allow full nodes to connect to validators for gossip
```

Comment:
I think it'd be worthwhile to extend this comment to also note that block authors will also not be considered parachain validators in the case that the maximum number of validators participating in parachain validation is limited.

Although...due to https://github.com/paritytech/polkadot/blob/master/runtime/parachains/src/runtime_api_impl/v1.rs#L243 we currently include all of the next authority IDs (even beyond `max_validators`) which are most likely the same as current-session.

Reply:
that's exactly the motivation for using authorities api instead of session_info

Reply:
Do you mind adding some of that info to the comment?

Reply:
Hmm, do we actually want to add all validators to the validation reserved peerset or just the ones participating in the parachain validation? If we add all, then we can set it to reserved only, no? @eskimor

Reply:
@ordian We had this discussion earlier today, and I believe the outcome was:
  * Connect to parachain validators of the last N sessions
  * Connect to all validators of the current session so block authors are included too.

Technically we don't need to include block authors as they should be reachable by `in_peers`  and `out_peers`, but this ensures they're only one hop away.

Reply:
> Technically we don't need to include block authors as they should be reachable by `in_peers` and `out_peers`, but this ensures they're only one hop away.

We can not rely on that, peer sets can easily get filled either maliciously or by accident. We also don't punish inactive peers, so that's a really cheap attack.

Reply:
Copying my DM from Element:

here's my understanding of what we want
* a fully connected graph of all validators with a clique of parachain validators, although the latter part is optional
* for disputes we want parachain validators of some previous sessions

We could make e.g. statement distribution and disputes use a separate peerset each and set reserved validation to only parachain validators.
But for now, using authorities is fine I think.

By using AuthorityDiscoveryApi::authorities we will set reserved peerset to a superset of that, but as you mentioned that's mostly two validator groups, also this will ensure we're potentially connected to validators in the next session in advance, which is nice